grammar.js

```diff
@@ -343,12 +344,52 @@ module.exports = grammar({
     },
 
     // LiteralString
-    string: ($) =>
-      seq(
-        field('start', alias($._string_start, 'string_start')),
-        field('content', optional(alias($._string_content, 'string_content'))),
-        field('end', alias($._string_end, 'string_end'))
-      ),
+    string: $ => choice($._string_literal, $._multiline_string_literal),
+
+    _string_literal: $ => choice(
+      seq('"', optional(alias($._unescaped_double_string_content, $.string_content)), '"'),
+      seq('\'', optional(alias($._unescaped_single_string_content, $.string_content)), '\''),
+    ),
+
+    _multiline_string_literal: $ => seq(
+      alias($._multiline_string_start, '[['),
+      alias($._multiline_string_content, $.string_content),
+      alias($._multiline_string_end, ']]'),
+    ),
+
+    // Workaround to https://github.com/tree-sitter/tree-sitter/issues/1156
+    // We give names to the token() constructs containing a regexp
+    // so as to obtain a node in the CST.
+    //
+    _unescaped_double_string_content_slice: _ => token.immediate(prec(1, /[^"\\]+/)),
+    _unescaped_double_string_content: $ => repeat1(choice(
+      $._unescaped_double_string_content_slice,
+      $._escape_sequence,
+    )),
+
+    // same here
+    _unescaped_single_string_content_slice: _ => token.immediate(prec(1, /[^'\\]+/)),
+    _unescaped_single_string_content: $ => repeat1(choice(
+      $._unescaped_single_string_content_slice,
+      $._escape_sequence,
+    )),
+
+    _escape_sequence: $ => choice(
+      prec(2, token.immediate(seq('\\', /[^abfnrtvxuz'\"\\\?]/))),
+      prec(1, $.escape_sequence),
+    ),
+
+    escape_sequence: _ => token.immediate(seq(
+      '\\',
+      choice(
+        /[^xu0-7]/,
+        /[0-9]{1,3}/,
+        /x[0-9a-fA-F]{2}/,
+        /u[0-9a-fA-F]{4}/,
```

Comment:
Just double checking... are these valid lua escaped string?

`/u[0-9a-fA-F]{4}/`, `/U[0-9a-fA-F]{8}/`

Reply:
I believe 5.3 added support for it

Reply:
I couldn't get them to work with any of Lua 5.x or LuaJIT.

Reply:
My mistake, it requires brackets, so `\u{0090}`